File: 49:GroupAnagrams.py



class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashMap = {}
        for char in strs:
            sortedChar = "".join(sorted(char))
            if sortedChar in hashMap:
                hashMap[sortedChar] += [char]
            else:
                hashMap[sortedChar] = [char]
        output = []
        for key in hashMap:
            output += [hashMap[key]]
        return output

File: 49:GroupAnagrams.java
//Leetcode Medium Problem

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> hM = new HashMap<>();

        for (String str : strs) {
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String key = new String(chars);

            hM.putIfAbsent(key, new ArrayList<String>());
            hM.get(key).add(str);
        }

        return new ArrayList<>(hM.values());
    }
}
